### ISSUE TITLE
Stale connections not being detected on flush prelude

### ISSUE THREAD

**Issue by rossabaker** [http4s/http4s#854]
If a request has an effectful body, the effect is run twice when we discard a stale connection and retry.

Running an echo server, like org.http4s.blaze.examples.NIO1HttpServer, on port 8080:

```
scala> var counter = 0
counter: Int = 0

scala> val c = client.blaze.PooledHttp1Client()
c: org.http4s.client.Client = Client(Kleisli(org.http4s.client.blaze.BlazeClient$$$Lambda$1115/1372291320@2c00a47e),scalaz.concurrent.Task@1990775f)

scala> def makeRequest = Request(uri = Uri.uri("http://localhost:8080/echo")).withBody(scalaz.stream.Process.eval(Task.delay { counter += 1; counter.toString }))
makeRequest: scalaz.concurrent.Task[org.http4s.Request]

scala> c.expect[String](makeRequest).run
res2: String = 1

scala> c.expect[String](makeRequest).run
res3: String = 2

scala> c.expect[String](makeRequest).run
res4: String = 3
```

Bounce server

```
scala> c.expect[String](makeRequest).run
res5: String = 5
```

`res5` should be 4.  The connection manager seems to be detecting an EOF, but it's not getting the EOF on flushPrelude.  In a separate reproduction, @SteveDraper reports hitting [this success case](https://github.com/http4s/http4s/blob/master/blaze-client/src/main/scala/org/http4s/client/blaze/Http1Connection.scala#L145).

**Comment by bryce-anderson** [comment]
I can reproduce this on linux (Ubuntu 16.04) as well. Apparently we were not as clever as we though, and flushing the prelude doesn't actually fail on a stale connection. :frowning:

**Comment by bryce-anderson** [comment]
The two interesting frames of the request sent on a stale connection captured via wireshark:
```

1	0.000000000	127.0.0.1	127.0.0.1	TCP	207	[TCP segment of a reassembled PDU]	51086	8080
2	0.000015604	127.0.0.1	127.0.0.1	TCP	54	8080 → 51086 [RST] Seq=1 Win=0 Len=0	8080	51086

// Data from the frame preceding the RST:
GET /echo HTTP/1.1
Host: localhost:8080
Content-Type: text/plain; charset=UTF-8
Accept: text/*
User-Agent: http4s-blaze/0.16.0-SNAPSHOT
```

Looking at the wireshark output while restarting the server shows a `FIN` packet is sent from the server and the client replies with an `ACK` in turn, but that doesn't stop it from trying to send the next request on the same socket (successfully, as far as http4s knows) only to get slapped with a `RST` packet.

**Comment by bryce-anderson** [comment]
Have a patch ready to fix this: https://github.com/http4s/http4s/pull/856

**Comment by bryce-anderson** [comment]
Closed via #856.
